File: backend/src/main/java/nl/wilcokas/luckystackworker/dto/ProfileDTO.java
package nl.wilcokas.luckystackworker.dto;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import nl.wilcokas.luckystackworker.model.ChannelEnum;
import nl.wilcokas.luckystackworker.model.PSF;
import nl.wilcokas.luckystackworker.model.Profile;
import nl.wilcokas.luckystackworker.service.dto.OpenImageModeEnum;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProfileDTO {

    public ProfileDTO(Profile profile) {
        this.name = profile.getName();

        // sharpen
        this.applySharpenToChannel = ChannelEnum.RGB;
        this.radius = profile.getRadius();
        this.amount = profile.getAmount();
        this.iterations = profile.getIterations();
        this.level = profile.getLevel();
        this.clippingStrength = profile.getClippingStrength();
        this.clippingRange = profile.getClippingRange();
        this.deringRadius = profile.getDeringRadius();
        this.deringStrength = profile.getDeringStrength();
        this.blendRaw = profile.getBlendRaw();
        this.wienerIterations = profile.getWienerIterations();

        this.radiusGreen = profile.getRadiusGreen();
        this.amountGreen = profile.getAmountGreen();
        this.iterationsGreen = profile.getIterationsGreen();
        this.levelGreen = profile.getLevelGreen();
        this.clippingStrengthGreen = profile.getClippingStrengthGreen();
        this.clippingRangeGreen = profile.getClippingRangeGreen();
        this.deringRadiusGreen = profile.getDeringRadiusGreen();
        this.deringStrengthGreen = profile.getDeringStrengthGreen();
        this.blendRawGreen = profile.getBlendRawGreen();
        this.wienerIterationsGreen = profile.getWienerIterationsGreen();

        this.radiusBlue = profile.getRadiusBlue();
        this.amountBlue = profile.getAmountBlue();
        this.iterationsBlue = profile.getIterationsBlue();
        this.levelBlue = profile.getLevelBlue();
        this.clippingStrengthBlue = profile.getClippingStrengthBlue();
        this.clippingRangeBlue = profile.getClippingRangeBlue();
        this.deringRadiusBlue = profile.getDeringRadiusBlue();
        this.deringStrengthBlue = profile.getDeringStrengthBlue();
        this.blendRawBlue = profile.getBlendRawBlue();
        this.wienerIterationsBlue = profile.getWienerIterationsBlue();

        this.sharpenMode = profile.getSharpenMode();
        this.luminanceIncludeRed = profile.isLuminanceIncludeRed();
        this.luminanceIncludeGreen = profile.isLuminanceIncludeGreen();
        this.luminanceIncludeBlue = profile.isLuminanceIncludeBlue();
        this.luminanceIncludeColor = profile.isLuminanceIncludeColor();
        this.applyWienerDeconvolution = profile.getApplyWienerDeconvolution();
        this.applyUnsharpMask = profile.getApplyUnsharpMask();

        // denoise
        this.denoiseAlgorithm1 = profile.getDenoiseAlgorithm1();
        this.denoise1Amount = profile.getDenoise1Amount();
        this.denoise1Radius = profile.getDenoise1Radius();
        this.denoise1Iterations = profile.getDenoise1Iterations();
        this.iansAmount = profile.getIansAmount();
        this.iansAmountMid = profile.getIansAmountMid();
        this.iansRecovery = profile.getIansRecovery();
        this.iansIterations = profile.getIansIterations();
        this.rofTheta = profile.getRofTheta();
        this.rofIterations = profile.getRofIterations();
        this.denoiseAlgorithm2 = profile.getDenoiseAlgorithm2();
        this.denoise2Radius = profile.getDenoise2Radius();
        this.denoise2Iterations = profile.getDenoise2Iterations();
        this.savitzkyGolaySize = profile.getSavitzkyGolaySize();
        this.savitzkyGolayAmount = profile.getSavitzkyGolayAmount();
        this.savitzkyGolayIterations = profile.getSavitzkyGolayIterations();
        this.bilateralIterations = profile.getBilateralIterations();
        this.bilateralRadius = profile.getBilateralRadius();
        this.bilateralSigmaColor = profile.getBilateralSigmaColor();
        this.bilateralSigmaSpace = profile.getBilateralSigmaSpace();

        this.denoise1AmountGreen = profile.getDenoise1AmountGreen();
        this.denoise1RadiusGreen = profile.getDenoise1RadiusGreen();
        this.denoise1IterationsGreen = profile.getDenoise1IterationsGreen();
        this.denoise2RadiusGreen = profile.getDenoise2RadiusGreen();
        this.denoise2IterationsGreen = profile.getDenoise2IterationsGreen();
        this.savitzkyGolaySizeGreen = profile.getSavitzkyGolaySizeGreen();
        this.savitzkyGolayAmountGreen = profile.getSavitzkyGolayAmountGreen();
        this.savitzkyGolayIterationsGreen = profile.getSavitzkyGolayIterationsGreen();
        this.rofThetaGreen = profile.getRofThetaGreen();
        this.rofIterationsGreen = profile.getRofIterationsGreen();
        this.bilateralIterationsGreen = profile.getBilateralIterationsGreen();
        this.bilateralRadiusGreen = profile.getBilateralRadiusGreen();
        this.bilateralSigmaColorGreen = profile.getBilateralSigmaColorGreen();
        this.bilateralSigmaSpaceGreen = profile.getBilateralSigmaSpaceGreen();

        this.denoise1AmountBlue = profile.getDenoise1AmountBlue();
        this.denoise1RadiusBlue = profile.getDenoise1RadiusBlue();
        this.denoise1IterationsBlue = profile.getDenoise1IterationsBlue();
        this.denoise2RadiusBlue = profile.getDenoise2RadiusBlue();
        this.denoise2IterationsBlue = profile.getDenoise2IterationsBlue();
        this.savitzkyGolaySizeBlue = profile.getSavitzkyGolaySizeBlue();
        this.savitzkyGolayAmountBlue = profile.getSavitzkyGolayAmountBlue();
        this.savitzkyGolayIterationsBlue = profile.getSavitzkyGolayIterationsBlue();
        this.rofThetaBlue = profile.getRofThetaBlue();
        this.rofIterationsBlue = profile.getRofIterationsBlue();
        this.bilateralIterationsBlue = profile.getBilateralIterationsBlue();
        this.bilateralRadiusBlue = profile.getBilateralRadiusBlue();
        this.bilateralSigmaColorBlue = profile.getBilateralSigmaColorBlue();
        this.bilateralSigmaSpaceBlue = profile.getBilateralSigmaSpaceBlue();

        // color & dispersion
        this.red = profile.getRed();
        this.green = profile.getGreen();
        this.blue = profile.getBlue();
        this.purple = profile.getPurple();
        this.saturation = profile.getSaturation();
        this.normalizeColorBalance = profile.isNormalizeColorBalance();

        // constrast & light
        this.gamma = profile.getGamma();
        this.contrast = profile.getContrast();
        this.brightness = profile.getBrightness();
        this.lightness = profile.getLightness();
        this.background = profile.getBackground();
        this.localContrastMode = profile.getLocalContrastMode();
        this.localContrastFine = profile.getLocalContrastFine();
        this.localContrastMedium = profile.getLocalContrastMedium();
        this.localContrastLarge = profile.getLocalContrastLarge();
        this.dispersionCorrectionEnabled = profile.isDispersionCorrectionEnabled();
        this.dispersionCorrectionRedX = profile.getDispersionCorrectionRedX();
        this.dispersionCorrectionBlueX = profile.getDispersionCorrectionBlueX();
        this.dispersionCorrectionRedY = profile.getDispersionCorrectionRedY();
        this.dispersionCorrectionBlueY = profile.getDispersionCorrectionBlueY();
        this.preserveDarkBackground = profile.isPreserveDarkBackground();
        this.gain = profile.getGain();

        this.scale = profile.getScale();
        this.rotationAngle = profile.getRotationAngle();
        this.equalizeLocalHistogramsStrength = profile.getEqualizeLocalHistogramsStrength();
        this.openImageMode = profile.getOpenImageMode();
        PSF profilePsf = profile.getPsf();
        this.psf = PSFDTO.builder()
                .airyDiskRadius(profilePsf.getAiryDiskRadius())
                .seeingIndex(profilePsf.getSeeingIndex())
                .diffractionIntensity(profilePsf.getDiffractionIntensity())
                .type(profilePsf.getType())
                .build();
    }

    private String name;

    // sharpen
    private BigDecimal radius;
    private BigDecimal amount;
    private int iterations;
    private int level;
    private int clippingStrength;
    private int clippingRange;
    private BigDecimal deringRadius;
    private int deringStrength;
    private int blendRaw;
    private int wienerIterations;
    private double gain;

    private BigDecimal radiusGreen;
    private BigDecimal amountGreen;
    private int iterationsGreen;
    private int levelGreen;
    private int clippingStrengthGreen;
    private int clippingRangeGreen;
    private BigDecimal deringRadiusGreen;
    private int deringStrengthGreen;
    private int blendRawGreen;
    private int wienerIterationsGreen;

    private BigDecimal radiusBlue;
    private BigDecimal amountBlue;
    private int iterationsBlue;
    private int levelBlue;
    private int clippingStrengthBlue;
    private int clippingRangeBlue;
    private BigDecimal deringRadiusBlue;
    private int deringStrengthBlue;
    private int blendRawBlue;
    private int wienerIterationsBlue;

    private String sharpenMode;
    private ChannelEnum applySharpenToChannel;
    private boolean applyUnsharpMask;
    private boolean applyWienerDeconvolution;

    // denoise
    private String denoiseAlgorithm1;
    private BigDecimal denoise1Amount;
    private BigDecimal denoise1Radius;
    private int denoise1Iterations;
    private BigDecimal iansAmount;
    private BigDecimal iansAmountMid;
    private BigDecimal iansRecovery;
    private int iansIterations;
    private int rofTheta;
    private int rofIterations;
    private String denoiseAlgorithm2;
    private int savitzkyGolaySize;
    private int savitzkyGolayAmount;
    private int savitzkyGolayIterations;
    private BigDecimal denoise2Radius;
    private int denoise2Iterations;
    private ChannelEnum applyDenoiseToChannel;
    private int bilateralSigmaColor;
    private int bilateralSigmaSpace;
    private int bilateralRadius;
    private int bilateralIterations;


    private BigDecimal denoise1AmountGreen;
    private BigDecimal denoise1RadiusGreen;
    private int denoise1IterationsGreen;
    private BigDecimal iansAmountGreen;
    private BigDecimal iansRecoveryGreen;
    private int rofThetaGreen;
    private int rofIterationsGreen;
    private int savitzkyGolaySizeGreen;
    private int savitzkyGolayAmountGreen;
    private int savitzkyGolayIterationsGreen;
    private BigDecimal denoise2RadiusGreen;
    private int denoise2IterationsGreen;
    private int bilateralSigmaColorGreen;
    private int bilateralSigmaSpaceGreen;
    private int bilateralRadiusGreen;
    private int bilateralIterationsGreen;

    private BigDecimal denoise1AmountBlue;
    private BigDecimal denoise1RadiusBlue;
    private int denoise1IterationsBlue;
    private BigDecimal iansAmountBlue;
    private BigDecimal iansRecoveryBlue;
    private int rofThetaBlue;
    private int rofIterationsBlue;
    private int savitzkyGolaySizeBlue;
    private int savitzkyGolayAmountBlue;
    private int savitzkyGolayIterationsBlue;
    private BigDecimal denoise2RadiusBlue;
    private int denoise2IterationsBlue;
    private int bilateralSigmaColorBlue;
    private int bilateralSigmaSpaceBlue;
    private int bilateralRadiusBlue;
    private int bilateralIterationsBlue;

    // light & contrast
    private BigDecimal gamma;
    private int contrast;
    private int brightness;
    private int lightness;
    private int background;
    private String localContrastMode;
    private int localContrastFine;
    private int localContrastMedium;
    private int localContrastLarge;
    private boolean preserveDarkBackground;

    // color & dispersion
    private BigDecimal red;
    private BigDecimal green;
    private BigDecimal blue;
    private BigDecimal purple;
    private BigDecimal saturation;
    private boolean dispersionCorrectionEnabled;
    private int dispersionCorrectionRedX;
    private int dispersionCorrectionBlueX;
    private int dispersionCorrectionRedY;
    private int dispersionCorrectionBlueY;
    private boolean luminanceIncludeRed;
    private boolean luminanceIncludeGreen;
    private boolean luminanceIncludeBlue;
    private boolean luminanceIncludeColor;
    private boolean normalizeColorBalance;

    private double scale;
    private double rotationAngle;
    private int equalizeLocalHistogramsStrength;
    private OpenImageModeEnum openImageMode;
    private PSFDTO psf;
}

File: backend/src/main/kotlin/nl/wilcokas/luckystackworker/dto/ProfileDTO.kt
package nl.wilcokas.luckystackworker.dto

import nl.wilcokas.luckystackworker.model.ChannelEnum
import nl.wilcokas.luckystackworker.model.Profile
import nl.wilcokas.luckystackworker.service.bean.OpenImageModeEnum
import java.math.BigDecimal

data class ProfileDTO(
    var name: String? = null,

    // sharpen
    var radius: BigDecimal? = null,
    var amount: BigDecimal? = null,
    var iterations: Int = 0,
    var level: Int = 0,
    var clippingStrength: Int = 0,
    var clippingRange: Int = 0,
    var deringRadius: BigDecimal? = null,
    var deringStrength: Int = 0,
    var blendRaw: Int = 0,
    var wienerIterations: Int = 0,
    var wienerRepetitions: Int = 0,
    var clippingSuppression: Double = 0.0,

    var radiusGreen: BigDecimal? = null,
    var amountGreen: BigDecimal? = null,
    var iterationsGreen: Int = 0,
    var levelGreen: Int = 0,
    var clippingStrengthGreen: Int = 0,
    var clippingRangeGreen: Int = 0,
    var deringRadiusGreen: BigDecimal? = null,
    var deringStrengthGreen: Int = 0,
    var blendRawGreen: Int = 0,
    var wienerIterationsGreen: Int = 0,

    var radiusBlue: BigDecimal? = null,
    var amountBlue: BigDecimal? = null,
    var iterationsBlue: Int = 0,
    var levelBlue: Int = 0,
    var clippingStrengthBlue: Int = 0,
    var clippingRangeBlue: Int = 0,
    var deringRadiusBlue: BigDecimal? = null,
    var deringStrengthBlue: Int = 0,
    var blendRawBlue: Int = 0,
    var wienerIterationsBlue: Int = 0,

    var sharpenMode: String? = null,
    var applySharpenToChannel: ChannelEnum? = null,
    var applyUnsharpMask: Boolean = false,
    var applyWienerDeconvolution: Boolean = false,

    // denoise
    var denoiseAlgorithm1: String? = null,
    var denoise1Amount: BigDecimal? = null,
    var denoise1Radius: BigDecimal? = null,
    var denoise1Iterations: Int = 0,
    var denoiseAlgorithm2: String? = null,
    var savitzkyGolaySize: Int = 0,
    var savitzkyGolayAmount: Int = 0,
    var savitzkyGolayIterations: Int = 0,
    var denoise2Radius: BigDecimal? = null,
    var denoise2Iterations: Int = 0,
    var applyDenoiseToChannel: ChannelEnum? = null,
    var bilateralSigmaColor: Int = 0,
    var bilateralSigmaSpace: Int = 0,
    var bilateralRadius: Int = 0,
    var bilateralIterations: Int = 0,

    var denoise1AmountGreen: BigDecimal? = null,
    var denoise1RadiusGreen: BigDecimal? = null,
    var denoise1IterationsGreen: Int = 0,
    var iansAmountGreen: BigDecimal? = null,
    var iansRecoveryGreen: BigDecimal? = null,
    var rofThetaGreen: Int = 0,
    var rofIterationsGreen: Int = 0,
    var savitzkyGolaySizeGreen: Int = 0,
    var savitzkyGolayAmountGreen: Int = 0,
    var savitzkyGolayIterationsGreen: Int = 0,
    var denoise2RadiusGreen: BigDecimal? = null,
    var denoise2IterationsGreen: Int = 0,
    var bilateralSigmaColorGreen: Int = 0,
    var bilateralSigmaSpaceGreen: Int = 0,
    var bilateralRadiusGreen: Int = 0,
    var bilateralIterationsGreen: Int = 0,

    var denoise1AmountBlue: BigDecimal? = null,
    var denoise1RadiusBlue: BigDecimal? = null,
    var denoise1IterationsBlue: Int = 0,
    var iansAmountBlue: BigDecimal? = null,
    var iansRecoveryBlue: BigDecimal? = null,
    var rofThetaBlue: Int = 0,
    var rofIterationsBlue: Int = 0,
    var savitzkyGolaySizeBlue: Int = 0,
    var savitzkyGolayAmountBlue: Int = 0,
    var savitzkyGolayIterationsBlue: Int = 0,
    var denoise2RadiusBlue: BigDecimal? = null,
    var denoise2IterationsBlue: Int = 0,
    var bilateralSigmaColorBlue: Int = 0,
    var bilateralSigmaSpaceBlue: Int = 0,
    var bilateralRadiusBlue: Int = 0,
    var bilateralIterationsBlue: Int = 0,

    // light & contrast
    var gamma: BigDecimal? = null,
    var contrast: Int = 0,
    var brightness: Int = 0,
    var lightness: Int = 0,
    var background: Int = 0,
    var localContrastMode: String? = null,
    var localContrastFine: Int = 0,
    var localContrastMedium: Int = 0,
    var localContrastLarge: Int = 0,
    var preserveDarkBackground: Boolean = false,

    // color & dispersion
    var red: BigDecimal? = null,
    var green: BigDecimal? = null,
    var blue: BigDecimal? = null,
    var purple: BigDecimal? = null,
    var saturation: BigDecimal? = null,
    var dispersionCorrectionEnabled: Boolean = false,
    var dispersionCorrectionRedX: Double = 0.0,
    var dispersionCorrectionBlueX: Double = 0.0,
    var dispersionCorrectionRedY: Double = 0.0,
    var dispersionCorrectionBlueY: Double = 0.0,
    var luminanceIncludeRed: Boolean = false,
    var luminanceIncludeGreen: Boolean = false,
    var luminanceIncludeBlue: Boolean = false,
    var luminanceIncludeColor: Boolean = false,
    var normalizeColorBalance: Boolean = false,

    var scale: Double = 0.0,
    var rotationAngle: Double = 0.0,
    var openImageMode: OpenImageModeEnum? = null,
    var psf: PSFDTO? = null

) {
    // Secondary constructor for mapping from domain Profile
    constructor(profile: Profile) : this() {
        this.name = profile.name

        // sharpen
        this.applySharpenToChannel = ChannelEnum.RGB
        this.radius = profile.radius
        this.amount = profile.amount
        this.iterations = profile.iterations
        this.level = profile.level
        this.clippingStrength = profile.clippingStrength
        this.clippingRange = profile.clippingRange
        this.deringRadius = profile.deringRadius
        this.deringStrength = profile.deringStrength
        this.blendRaw = profile.blendRaw
        this.wienerIterations = profile.wienerIterations
        this.wienerRepetitions = profile.wienerRepetitions

        this.radiusGreen = profile.radiusGreen
        this.amountGreen = profile.amountGreen
        this.iterationsGreen = profile.iterationsGreen
        this.levelGreen = profile.levelGreen
        this.clippingStrengthGreen = profile.clippingStrengthGreen
        this.clippingRangeGreen = profile.clippingRangeGreen
        this.deringRadiusGreen = profile.deringRadiusGreen
        this.deringStrengthGreen = profile.deringStrengthGreen
        this.blendRawGreen = profile.blendRawGreen
        this.wienerIterationsGreen = profile.wienerIterationsGreen

        this.radiusBlue = profile.radiusBlue
        this.amountBlue = profile.amountBlue
        this.iterationsBlue = profile.iterationsBlue
        this.levelBlue = profile.levelBlue
        this.clippingStrengthBlue = profile.clippingStrengthBlue
        this.clippingRangeBlue = profile.clippingRangeBlue
        this.deringRadiusBlue = profile.deringRadiusBlue
        this.deringStrengthBlue = profile.deringStrengthBlue
        this.blendRawBlue = profile.blendRawBlue
        this.wienerIterationsBlue = profile.wienerIterationsBlue

        this.sharpenMode = profile.sharpenMode
        this.luminanceIncludeRed = profile.luminanceIncludeRed
        this.luminanceIncludeGreen = profile.luminanceIncludeGreen
        this.luminanceIncludeBlue = profile.luminanceIncludeBlue
        this.luminanceIncludeColor = profile.luminanceIncludeColor

        // Let op: In Profile was dit Boolean object (nullable), hier primitive boolean.
        // We gebruiken Elvis operator ?: false voor safety
        this.applyWienerDeconvolution = profile.applyWienerDeconvolution ?: false
        this.applyUnsharpMask = profile.applyUnsharpMask ?: false

        // denoise
        this.denoiseAlgorithm1 = profile.denoiseAlgorithm1
        this.denoise1Amount = profile.denoise1Amount
        this.denoise1Radius = profile.denoise1Radius
        this.denoise1Iterations = profile.denoise1Iterations
        this.denoiseAlgorithm2 = profile.denoiseAlgorithm2
        this.denoise2Radius = profile.denoise2Radius
        this.denoise2Iterations = profile.denoise2Iterations
        this.savitzkyGolaySize = profile.savitzkyGolaySize
        this.savitzkyGolayAmount = profile.savitzkyGolayAmount
        this.savitzkyGolayIterations = profile.savitzkyGolayIterations
        this.bilateralIterations = profile.bilateralIterations
        this.bilateralRadius = profile.bilateralRadius
        this.bilateralSigmaColor = profile.bilateralSigmaColor
        this.bilateralSigmaSpace = profile.bilateralSigmaSpace

        this.denoise1AmountGreen = profile.denoise1AmountGreen
        this.denoise1RadiusGreen = profile.denoise1RadiusGreen
        this.denoise1IterationsGreen = profile.denoise1IterationsGreen
        this.denoise2RadiusGreen = profile.denoise2RadiusGreen
        this.denoise2IterationsGreen = profile.denoise2IterationsGreen
        this.savitzkyGolaySizeGreen = profile.savitzkyGolaySizeGreen
        this.savitzkyGolayAmountGreen = profile.savitzkyGolayAmountGreen
        this.savitzkyGolayIterationsGreen = profile.savitzkyGolayIterationsGreen
        this.rofThetaGreen = profile.rofThetaGreen
        this.rofIterationsGreen = profile.rofIterationsGreen
        this.bilateralIterationsGreen = profile.bilateralIterationsGreen
        this.bilateralRadiusGreen = profile.bilateralRadiusGreen
        this.bilateralSigmaColorGreen = profile.bilateralSigmaColorGreen
        this.bilateralSigmaSpaceGreen = profile.bilateralSigmaSpaceGreen

        this.denoise1AmountBlue = profile.denoise1AmountBlue
        this.denoise1RadiusBlue = profile.denoise1RadiusBlue
        this.denoise1IterationsBlue = profile.denoise1IterationsBlue
        this.denoise2RadiusBlue = profile.denoise2RadiusBlue
        this.denoise2IterationsBlue = profile.denoise2IterationsBlue
        this.savitzkyGolaySizeBlue = profile.savitzkyGolaySizeBlue
        this.savitzkyGolayAmountBlue = profile.savitzkyGolayAmountBlue
        this.savitzkyGolayIterationsBlue = profile.savitzkyGolayIterationsBlue
        this.rofThetaBlue = profile.rofThetaBlue
        this.rofIterationsBlue = profile.rofIterationsBlue
        this.bilateralIterationsBlue = profile.bilateralIterationsBlue
        this.bilateralRadiusBlue = profile.bilateralRadiusBlue
        this.bilateralSigmaColorBlue = profile.bilateralSigmaColorBlue
        this.bilateralSigmaSpaceBlue = profile.bilateralSigmaSpaceBlue

        // color & dispersion
        this.red = profile.red
        this.green = profile.green
        this.blue = profile.blue
        this.purple = profile.purple
        this.saturation = profile.saturation
        this.normalizeColorBalance = profile.normalizeColorBalance

        // constrast & light
        this.gamma = profile.gamma
        this.contrast = profile.contrast
        this.brightness = profile.brightness
        this.lightness = profile.lightness
        this.background = profile.background
        this.localContrastMode = profile.localContrastMode
        this.localContrastFine = profile.localContrastFine
        this.localContrastMedium = profile.localContrastMedium
        this.localContrastLarge = profile.localContrastLarge
        this.dispersionCorrectionEnabled = profile.dispersionCorrectionEnabled
        this.dispersionCorrectionRedX = profile.dispersionCorrectionRedX
        this.dispersionCorrectionBlueX = profile.dispersionCorrectionBlueX
        this.dispersionCorrectionRedY = profile.dispersionCorrectionRedY
        this.dispersionCorrectionBlueY = profile.dispersionCorrectionBlueY
        this.preserveDarkBackground = profile.preserveDarkBackground
        this.clippingSuppression = profile.clippingSuppression

        this.scale = profile.scale
        this.rotationAngle = profile.rotationAngle
        this.openImageMode = profile.openImageMode

        val profilePsf = profile.psf
        if (profilePsf != null) {
            this.psf = PSFDTO(
                profilePsf.airyDiskRadius,
                profilePsf.seeingIndex,
                profilePsf.diffractionIntensity,
                profilePsf.type
            )
        }
    }
}